### PR TITLE
Add support for viewing 'User-supplied values' of helm release

### LINKED ISSUE #553: Add value diff between helm release versions and with defaults
**What would you like to be added**:
* [ ] A diff between default values and user values
* [ ] A diff between old and new values between releases

**Why is this needed**:
That would provide easier insight as to what the user changed from the default
and what has changed between versions of the charts

**Environment you are Lens application on:**

- Kubernetes distribution: vanilla v1.18.3 "on-premise"
- Desktop OS: windows & archlinux

### PR DESCRIPTION
**Pull request by osddeitf**
### What's included?
- Add checkbox to view "User-supplied values" instead of "merged with default values" of Helm release
- Essentially toggle `--all` from `helm get values --all -o yaml -n <namespace> <release>`.
### Motivation
- For readability + improve inline edit when lots of configurations are available.
- Reduce the risks when upgrade between Helm chart versions.
- Fix: #553

### CODE REVIEW

File: src/renderer/api/endpoints/helm-releases.api.ts
Comment (Nokel81) [inline src/renderer/api/endpoints/helm-releases.api.ts]:
```suggestion
    const path = `${endpoint({ name, namespace })}/values${all ? "?all" : ""}`;
```

File: src/main/helm/helm-release-manager.ts
Comment (Nokel81) [inline src/main/helm/helm-release-manager.ts:96]:
```suggestion
    const { stdout,  } = await promiseExec(`"${helm}" get values ${name} ${all ? "--all" : ""} --output yaml --namespace ${namespace} --kubeconfig ${pathToKubeconfig}`).catch((error) => { throw(error.stderr);});
```

File: src/renderer/components/+apps-releases/release-details.tsx
Comment (Nokel81) [inline src/renderer/components/+apps-releases/release-details.tsx:137]: I don't think the checkbox should be visible while the values are loading.

Comment (osddeitf) [reply src/renderer/components/+apps-releases/release-details.tsx:137]: I disabled it, what about hiding it make a difference?

File: src/renderer/api/endpoints/helm-releases.api.ts
Comment (osddeitf) [reply src/renderer/api/endpoints/helm-releases.api.ts]: Are the spaces matter? It's already passed linting.

File: src/main/helm/helm-release-manager.ts
Comment (osddeitf) [reply src/main/helm/helm-release-manager.ts:96]: Same question as above.

**Review by Nokel81 — DISMISSED** [review]
This does seem to work as intended. I tried to see if the output in the editor ever changes but I don't know enough about helm to verify that.

**Comment by osddeitf** [comment]
> This does seem to work as intended. I tried to see if the output in the editor ever changes but I don't know enough about helm to verify that.

Thank you for your approval.
If you want to see what different, try install nginx with helm, like this:
```
kubectl create ns nginx-example
helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
helm repo update
helm install -n nginx-example ingress-nginx ingress-nginx/ingress-nginx
```
After you're done, just remove the namespace `nginx-example` to clean up.

**Comment by osddeitf** [comment]
Sorry, but any update on this?

**Comment by Nokel81** [comment]
I'll see about getting another review so that we can get this into the next release.

File: src/renderer/components/+apps-releases/release-details.tsx
Comment (jim-docker) [inline src/renderer/components/+apps-releases/release-details.tsx]: To keep current default behaviour this should be false initially

**Review by jim-docker — COMMENTED** [review]
@osddeitf Can you also rebase to latest master and recheck it works as expected?

File: src/renderer/components/+apps-releases/release-details.tsx
Comment (osddeitf) [reply src/renderer/components/+apps-releases/release-details.tsx]: Sure, I think it's stale by now, I'll update it soon when I have time, maybe this weekend. And after that could you review it for me soon so this could become upstream?

Comment (Nokel81) [reply src/renderer/components/+apps-releases/release-details.tsx]: I can do these changes if you would like and don't mind me force pushing to your branch

Comment (osddeitf) [reply src/renderer/components/+apps-releases/release-details.tsx]: Ok, I appreciate your help, I'm busy right now so I'm couldn't

**Comment by Nokel81** [comment]
Going to mark this as 4.2.1 as we weren't able to get this into an RC and it is a feature change.

**Comment by Nokel81** [comment]
@jim-docker This is what I have come up with in terms of loading animation.

https://user-images.githubusercontent.com/8225332/111816623-6e8a1a00-88b3-11eb-9f17-51435b8c8da2.mov

**Comment by jim-docker** [comment]
> @jim-docker This is what I have come up with in terms of loading animation.
> 

Nice, but I was only looking for `userSuppliedOnly` to be false :). I think the changes to `AceEditor` should come in a different PR.

**Comment by Nokel81** [comment]
@jim-docker done, and opened #2367 as a follow up (don't bother reviewing that until this has been merged).

**Review by Nokel81 — DISMISSED** [review]

**Review by jim-docker — DISMISSED** [review]

**Comment by osddeitf** [comment]
Thank you guys for helping me with this PR. I'm really appreciated.

**Review by Nokel81 — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Pull request merged**
